<?xml version="1.0" encoding="UTF-8"?>
<class xmlns="http://xml.phpdox.net/src" full="Psy\CodeCleaner\CallTimePassByReferencePass" namespace="Psy\CodeCleaner" name="CallTimePassByReferencePass" abstract="false" final="false" start="27" end="52">
  <file path="/home/robert/Desktop/Hogeschool/2ejaars Development/DevBOpdrachten/project56/http/laravel/vendor/psy/psysh/src/Psy/CodeCleaner" file="CallTimePassByReferencePass.php" realpath="/home/robert/Desktop/Hogeschool/2ejaars Development/DevBOpdrachten/project56/http/laravel/vendor/psy/psysh/src/Psy/CodeCleaner/CallTimePassByReferencePass.php" size="1419" time="2015-12-17T18:44:10+00:00" unixtime="1450377850" sha1="6bae46dfde17f95263e6098e597ed429911e43cb" relative="vendor/psy/psysh/src/Psy/CodeCleaner/CallTimePassByReferencePass.php"/>
  <docblock>
    <description compact="Validate that the user did not use the call-time pass-by-reference that causes a fatal error.">As of PHP 5.4.0, call-time pass-by-reference was removed, so using it will raise a fatal error.</description>
    <author value="Martin HasoÅˆ &lt;martin.hason@gmail.com&gt;"/>
  </docblock>
  <extends full="Psy\CodeCleaner\CodeCleanerPass" namespace="Psy\CodeCleaner" name="CodeCleanerPass"/>
  <method name="enterNode" start="36" end="51" abstract="false" final="false" static="false" visibility="public">
    <docblock>
      <description compact="Validate of use call-time pass-by-reference."/>
      <throws description="if the user used call-time pass-by-reference in PHP &gt;= 5.4.0" type="object">
        <type full="Psy\CodeCleaner\RuntimeException" namespace="Psy\CodeCleaner" name="RuntimeException"/>
      </throws>
      <param type="object" variable="$node">
        <type full="PhpParser\Node" namespace="PhpParser" name="Node"/>
      </param>
    </docblock>
    <parameter name="node" byreference="false" type="object">
      <type full="PhpParser\Node" namespace="PhpParser" name="Node"/>
    </parameter>
  </method>
</class>
